source/common/http/conn_manager_impl.cc

```diff
@@ -419,7 +419,7 @@ void ConnectionManagerImpl::ActiveStream::resetIdleTimer() {
 void ConnectionManagerImpl::ActiveStream::onIdleTimeout() {
   connection_manager_.stats_.named_.downstream_rq_idle_timeout_.inc();
   // If headers have not been sent to the user, send a 408.
-  if (response_headers_ != nullptr) {
+  if (request_headers_ == nullptr || response_headers_ != nullptr) {
```

Comment:
I think it's actually OK to send a reply if there are no request headers, we just can't do the gRPC check. I think this is what we want since you would still get a 408 request timeout response.

Reply:
Yes, I can do this, but we need to do a bit more plumbing, since `sendLocalReply()` doesn't work out of the box today for responses sent before request headers (e.g. it needs them to detect upgrades).

Reply:
Oh, alright. Hmm. Probably not worth it. Add a comment maybe?